gnomad_constraint/constraint/constraint_pipeline.py

```diff
@@ -44,6 +50,53 @@ def main(args):
         hl.read_table(get_processed_ht_path("exomes")), trimers=trimers
     )
 
+    context_ht = full_context_ht.filter(full_context_ht.locus.in_autosome_or_par())
+    exome_ht = full_exome_ht.filter(full_exome_ht.locus.in_autosome_or_par())
+    mutation_ht = hl.read_table(mutation_rate_ht_path).select("mu_snp")
+
+    context_x_ht = hl.filter_intervals(full_context_ht, [hl.parse_locus_interval("X")])
+    context_x_ht = context_x_ht.filter(context_x_ht.locus.in_x_nonpar())
+    context_y_ht = hl.filter_intervals(full_context_ht, [hl.parse_locus_interval("Y")])
+    context_y_ht = context_y_ht.filter(context_y_ht.locus.in_y_nonpar())
+
+    exome_x_ht = hl.filter_intervals(full_exome_ht, [hl.parse_locus_interval("X")])
+    exome_x_ht = exome_x_ht.filter(exome_x_ht.locus.in_x_nonpar())
+    exome_y_ht = hl.filter_intervals(full_exome_ht, [hl.parse_locus_interval("Y")])
+    exome_y_ht = exome_y_ht.filter(exome_y_ht.locus.in_y_nonpar())
+
+    if args.create_training_set:
+        get_proportion_observed_by_coverage(
+            exome_ht,
+            context_ht,
+            mutation_ht,
+            possible_file_path,
+            True,
+            args.dataset,
+            not args.skip_af_filter_upfront,
+        ).write(training_ht_path, overwrite=args.overwrite)
+        hl.read_table(training_ht_path).export(
+            training_ht_path.replace(".ht", ".txt.bgz")
+        )
+        get_proportion_observed_by_coverage(
+            exome_x_ht,
+            context_x_ht,
+            mutation_ht,
+            possible_file_path,
+            True,
+            args.dataset,
+            not args.skip_af_filter_upfront,
```

Comment:
If you take my suggestions these change to:
```suggestion
            impose_high_af_cutoff_upfront=not skip_af_filter_upfront,
            freq_expr=exome_x_ht.freq[dataset_freq_idx],
            max_af=max_af,
```